src/mkfs.fat.c

```diff
@@ -1569,6 +1572,14 @@ int main(int argc, char **argv)
 	    }
 	    break;
 
+    case OPT_OFFSET:
+        part_offset = strtoul(optarg, &tmp, 0);
+        if (*tmp) {
```

Comment:
Hi! This is not fully correct error handling of `strtoul` function. It e.g. passes on empty string on overflow/underflow values. More safe usage of strtoul:
```c
tmp = NULL;
errno = 0;
part_offset = strtoul(optarg, &tmp, 0);
if (!*optarg || *tmp || errno) {
```

Reply:
And probably you should use `strtoull` function to allow specifying >= 4GB offset on 32bit systems. I hope that `off_t` type is already 64bit.

Reply:
Fixed.

Reply:
You are missing resetting errno to zero prior to `strtoul` call. `strtoul` does not change errno when there is no error.

Reply:
Fixed. Thanks.